v1/ast/parser_test.go

```diff
@@ -70,138 +70,1710 @@ func TestNumberTerms(t *testing.T) {
+				})
+			}
+		})
+	}
+
+	t.Run("v0 with future keywords", func(t *testing.T) {
+		caps := CapabilitiesForThisVersion(CapabilitiesRegoVersion(RegoV0))
+		popts := ParserOptions{RegoVersion: RegoV0, Capabilities: caps}
+
+		for kw := range futureKeywordsV0 {
+			popts.FutureKeywords = []string{kw}
+
+			t.Run(kw, func(t *testing.T) {
+				input := "foo." + kw
+				exp := RefTerm(VarTerm("foo"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = "input." + kw
+				exp = RefTerm(VarTerm("input"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = "data." + kw
+				exp = RefTerm(VarTerm("data"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = fmt.Sprintf("data.%s.foo", kw)
+				exp = RefTerm(VarTerm("data"), StringTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = fmt.Sprintf(`data.%s["foo"]`, kw)
+				exp = RefTerm(VarTerm("data"), StringTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = "data.foo." + kw
+				exp = RefTerm(VarTerm("data"), StringTerm("foo"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = `data["foo"].` + kw
+				exp = RefTerm(VarTerm("data"), StringTerm("foo"), StringTerm(kw))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = kw + ".foo"
+				exp = RefTerm(VarTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+
+				input = kw + `["foo"]`
+				exp = RefTerm(VarTerm(kw), StringTerm("foo"))
+				assertParseOneTerm(t, input, input, exp, popts)
+			})
+		}
+	})
+}
+
+func dropCapabilityFeature(caps *Capabilities, feature string) *Capabilities {
+	feats := make([]string, 0, len(caps.Features))
+	for _, f := range caps.Features {
+		if f != feature {
+			feats = append(feats, f)
+		}
+	}
+	caps.Features = feats
+	return caps
+}
+
+func TestRefTermsContainingKeywords_NoCapability(t *testing.T) {
+	for _, regoVersion := range []RegoVersion{RegoV0, RegoV1} {
+		caps := CapabilitiesForThisVersion(CapabilitiesRegoVersion(regoVersion))
+		caps = dropCapabilityFeature(caps, FeatureKeywordsInRefs)
+		popts := ParserOptions{RegoVersion: regoVersion, Capabilities: caps}
+
+		t.Run(regoVersion.String(), func(t *testing.T) {
+			for _, kw := range KeywordsForRegoVersion(regoVersion) {
+				t.Run(kw, func(t *testing.T) {
+					input := "foo." + kw
+					expErr := fmt.Sprintf(`rego_parse_error: unexpected %s keyword: expected identifier
+	foo.%s
+	    ^`, kw, kw)
+					t.Run(input, func(t *testing.T) {
+						assertParseErrorContains(t, input, input, expErr, popts)
+					})
+
+					input = "input." + kw
+					expErr = fmt.Sprintf(`rego_parse_error: unexpected %s keyword: expected identifier
+	input.%s
+	      ^`, kw, kw)
+					t.Run(input, func(t *testing.T) {
+						assertParseErrorContains(t, input, input, expErr, popts)
+					})
```

Comment:
Huh interesting pattern. Not sure I've ever seen `t.Run()` used like this 😃 
```suggestion	
					t.Run("foo." + kw, func(t *testing.T) {
						expErr := fmt.Sprintf(`rego_parse_error: unexpected %s keyword: expected identifier
	foo.%s
	    ^`, kw, kw)
						assertParseErrorContains(t, input, input, expErr, popts)
					})

					t.Run("input." + kw, func(t *testing.T) {
						expErr = fmt.Sprintf(`rego_parse_error: unexpected %s keyword: expected identifier
	input.%s
	      ^`, kw, kw)
						assertParseErrorContains(t, input, input, expErr, popts)
					})
```
This is what would have kept the eyebrow down 😁

Reply:
Yeah true. We could piggy-back on [t.Name()](https://pkg.go.dev/testing#T.Name) but it would require us to split the string again.... fine as-is. I'll get used to it 😉